src/snowflake/cli/api/identifiers.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import re
+
+from click import ClickException
+from snowflake.cli.api.cli_global_context import cli_context
+from snowflake.cli.api.exceptions import FQNInconsistencyError, FQNNameError
+from snowflake.cli.api.project.schemas.identifier_model import ObjectIdentifierBaseModel
+from snowflake.cli.api.project.util import VALID_IDENTIFIER_REGEX, unquote_identifier
+
+
+class FQN:
```

Comment:
I like this. Any plans to make it work for stages too?

Reply:
Good question, I think this would required for new schema version, where `name` / `stage` can be either a string or object. Something like:

```yml
streamlit:
  - name: "foo"
  - stage:
       name: "stage"
       schema: "bar"


streamlit: 
    name: 
        name: "foo"
        schema: "bar"
```

(The name:name probably needs more thought)

Reply:
Yeah, we can cross that bridge if/when we get to it, was just curious.